application/src/main/java/org/togetherjava/tjbot/features/basic/Starboard.java

```diff
@@ -20,8 +30,17 @@ public Starboard(Config config) {
     @Override
     public void onMessageReactionAdd(@NotNull MessageReactionAddEvent event) {
         String emojiName = event.getReaction().getEmoji().asCustom().getName();
-        MessageChannel channel = event.getChannel();
-        // TODO
-
+        Guild guild = event.getGuild();
+        GuildChannel channel = event.getGuildChannel();
+        if (!config.getEmojiNames().contains(emojiName) || !guild.getPublicRole().hasPermission(channel, Permission.VIEW_CHANNEL)) {
+            return;
+        }
+        Optional<TextChannel> starboardChannel = guild.getTextChannelsByName(config.getStarboardChannelName(), false).stream().findFirst();
+        if (starboardChannel.isEmpty()) {
+            //TODO log?
```

Comment:
yes, warning level with explanation how to deactivate the feature (empty list of emojis in config)